### PR TITLE
/Logger for DB connection

### LINKED ISSUE #2: Implement Logging for Database Connection Status
**Overview**
In our current server setup, we utilize Prisma as the ORM to manage PostgreSQL database connections through a connection pool. However, our existing configuration does not include a feedback mechanism for actively monitoring the status of these database connections. This lack of visibility into real-time database connectivity complicates troubleshooting and limits our ability to swiftly detect and address connection issues. To enhance our system’s reliability and improve operational transparency, it is essential to implement a logging mechanism that can provide detailed feedback on the connection status.

**Task:**

- [ ] Integrate Connection Status Logging with Prisma:

Implement a logging system that works in conjunction with our Prisma setup to monitor and record the status of database connections. 

- [ ] Develop Detailed Log Outputs:

Ensure that the logs generated are informative and accessible, detailing successful connection events as well as any errors or failures. This detailed information will be crucial for effective troubleshooting and quick resolution of issues.

### PR DESCRIPTION
**Pull request by miabu-pashh**
# Pull Request Template

**Fixes #2 r**

### What was changed?
This update implements logging for the database connection in the Donor form file. It includes detailed logging of the Prisma database connection status in the console, enabling better monitoring during application startup.

### Notable Changes:

Added logging functionality for the Prisma connection in index.ts.
This ensures the connection to the database is logged in the console, allowing real-time monitoring during application startup.

### Why was it changed?

This enhancement improves the visibility of the database connection process during the build and runtime phases of the application. By logging the Prisma connection details, we can ensure that automatic migrations are successfully applied and that the database is connected and running properly on the backend.

This is crucial for identifying and diagnosing connection issues early, especially in production environments.

### How was it changed?

The database connection function has been integrated into the index.ts file, where it logs the connection status to the console. This is triggered whenever the application starts, providing real-time feedback on the status of the Prisma connection.

Key updates include:

Logging Prisma connection details during application startup.
Ensuring that any issues with database connectivity are immediately visible via the console.

### Screenshots that show the changes (if applicable):

image

____

image

### CODE REVIEW

**Comment by Anjali0407-git** [comment]
@miabu-pashh Changes looks good. I will test it and let you know

File: server/src/index.ts [hunk 1]
```diff
@@ -5,6 +5,9 @@ import cookieParser from 'cookie-parser';
 import logger from 'morgan';
 import cors from 'cors';
 
+import { PrismaClient } from '@prisma/client'; // Import Prisma
+const prisma = new PrismaClient(); // Initialize Prisma Client
+
 // import indexRouter from './routes/index';
```
Comment (saikiran0405) [inline server/src/index.ts:11]: It's good to remove unwanted code

File: server/src/index.ts [hunk 2]
```diff
@@ -47,9 +51,35 @@ app.use((err: any, req: Request, res: Response, next: NextFunction) => {
...
-const port = process.env.PORT || 5000; // Use environment variable or default to 3001
-app.listen(port, () => {
-  console.log(`Server running on http://localhost:${port}`);
+// const port = process.env.PORT || 3001; // Use environment variable or default to 3001
+// app.listen(port, () => {
+//   console.log(`Server running on http://localhost:${port}`);
+// });
+
+// Logging the DB connection and starting server
+const startServer = async () => {
```
Comment (saikiran0405) [inline server/src/index.ts:60]: In your catch block, you are logging generic error messages.
Possible solution: We can add more context to the error logs, such as timestamps or details about which part of the code failed.

File: server/src/index.ts [hunk 3]
```diff
@@ -47,9 +51,35 @@ app.use((err: any, req: Request, res: Response, next: NextFunction) => {
...
+    // Connect to the database
+    await prisma.$connect(); 
+    console.log('The logger: Connected to the database successfully!'); 
+
+    // Start the server
+    const port = process.env.PORT || 3001;
+    app.listen(port, () => {
+      console.log(`Server running on http://localhost:${port}`);
+    });
+  } catch (error) {
```
Comment (saikiran0405) [inline server/src/index.ts:71]: In your catch block, you are logging generic error messages.
Possible solution: We can add more context to the error logs, such as timestamps or details about which part of the code failed.

**Comment by Anjali0407-git** [comment]
@miabu-pashh I have finished testing your update and it correctly logs db connection status to the user. I have requested few suggestions to work on.

File: client-app/src/Components/Login.js
Comment (kate-holdener) [inline client-app/src/Components/Login.js:1]: Looks like you didn't make any meaningful changes in this file (add a blank line/remove a blank line). Please don't commit files that didn't need to be changed.

File: server/src/index.ts [hunk 4]
```diff
@@ -47,9 +51,35 @@ app.use((err: any, req: Request, res: Response, next: NextFunction) => {
   res.render('error');
 });
 
-const port = process.env.PORT || 5000; // Use environment variable or default to 3001
-app.listen(port, () => {
-  console.log(`Server running on http://localhost:${port}`);
+// const port = process.env.PORT || 3001; // Use environment variable or default to 3001
+// app.listen(port, () => {
+//   console.log(`Server running on http://localhost:${port}`);
+// });
```
Comment (kate-holdener) [inline server/src/index.ts:57]: Please don't leave commented out code in your final solution. Remove the code that's not needed.

File: server/src/routes/donorRoutes.ts
Comment (kate-holdener) [inline server/src/routes/donorRoutes.ts:1]: Looks like you didn't really change this file (added a blank line). Please don't commit files that you didn't intentionally modify.

File: client-app/src/Components/Login.js
Comment (miabu-pashh) [reply client-app/src/Components/Login.js:1]: Yeah, understood. Thanks for the update.

File: server/src/index.ts (hunk 4 above)
Comment (miabu-pashh) [reply server/src/index.ts:57]: yes, thank you. This code smells, I haven't taken it under consideration while pushing the code, It was before our recent class. Thanks for the update The next pr will not contain such activity.

**Comment by miabu-pashh** [comment]
I have implemented the changes that have been discussed and updated the code.

image

**Comment by Anjali0407-git** [comment]
@miabu-pashh some review comments were left unaddressed. Please address them

**Review by Anjali0407-git — APPROVED** [review]

**Pull request merged**